base/test.jl

```diff
@@ -271,6 +276,17 @@ if an exception is thrown inside the test block but outside of a `@test` context
 function record end
 
 """
+    should_run(ts::AbstractTestSet, nruns::Int)
+
+Decide if the tests should be run; returns true if and only if they should.
+The tests are called in a loop and the `nruns` arg counts the number of times the test
+block has been executed. `nruns` is 0 when the tests have not yet executed.
+"""
+function should_run(ts::AbstractTestSet, nruns::Int)
```

Comment:
I updated docs to indicate that implementing this one is optional. But maybe it was another comment that lead you to this? I think it is ok in the comment that introduces the interface since it says that the interface is defined by these three methods; to me this does not indicate it must be implemented.